modules/@apostrophecms/video-widget/ui/public/video.js

```diff
@@ -53,13 +56,28 @@ apos.util.widgetPlayers['@apostrophecms/video'] = {
         // video aspect ratio right
         if (result.width && result.height) {
           inner.style.width = '100%';
-          inner.style.height = ((result.height / result.width) * inner.offsetWidth) + 'px';
+          resizeVideo(inner);
+          // If we need to iniitally size the video, also resize it on window resize.
+          window.addEventListener('resize', resizeHandler);
         } else {
           // No, so assume the oembed HTML code is responsive.
         }
       }, 0);
     }
 
+    function resizeVideo(canvasEl) {
+      canvasEl.style.height = ((queryResult.height / queryResult.width) * canvasEl.offsetWidth) + 'px';
+    };
+
+    function resizeHandler() {
+      const canvas = el.querySelector('[data-apos-video-canvas]');
```

Comment:
This tests whether the youtube player is still inside the widget. I don't see how that could change. My concern is that the entire widget may no longer be in the DOM (for instance, because it has been deleted).